src/wired/error_boundry.tsx

```diff
@@ -1,32 +1,52 @@
 import Error, { ErrorProps } from 'next/error';
 import React, { Component, PropsWithChildren } from 'react';
+import { NextPageContext } from 'next';
 
 export type WiredErrorBoundaryProps = PropsWithChildren<{
-  ErrorComponent?: React.ComponentType<ErrorProps>;
+  ErrorComponent?: React.ComponentType<
+    ErrorProps & { error: NextPageContext['err'] }
+  >;
 }>;
 
 interface WiredErrorBoundaryState {
-  hasError: boolean;
+  error?: NextPageContext['err'];
 }
 
 export class WiredErrorBoundary extends Component<
   WiredErrorBoundaryProps,
   WiredErrorBoundaryState
 > {
-  static getDerivedStateFromError(): WiredErrorBoundaryState {
-    return { hasError: true };
+  static getDerivedStateFromError(
+    error: NextPageContext['err']
+  ): WiredErrorBoundaryState {
+    return { error };
   }
 
-  state = { hasError: false };
-
   render() {
     const ErrorComponent = this.props.ErrorComponent;
 
-    if (this.state.hasError) {
+    // Something happened, let consumers decide what to do with this type of error
+    let error: NextPageContext['err'];
+
+    // Because passing this.state.error won't work
+    if (this.state.error)
+      error = {
+        name: 'Error 500',
+        message: this.state.error ? this.state.error.toString() : 'Unknown',
+        stack: this.state.error?.stack
+          ? this.state.error.stack.toString()
+          : 'Unknown',
+        statusCode: 500,
+      };
+
+    // eslint-disable-next-line no-console
+    console.log('error_boundary:error', error);
```

Comment:
Please remove `console.log`

Reply:
Whoops, sorry. At least you know this was tested lol. Moving at 100mph today. Just pushed a commit removing it 🤪